### ISSUE TITLE
Nothing happens when used

### ISSUE THREAD

**Issue by luerhard** [UB-Mannheim/zotero-ocr#4]
Hey devs,
first of all: Nice idea to implement OCR for zotero! I have a problem with it: I installed the self-built .xpi-File (based on commit `63dacf0`) in zotero (Desktop version `5.0.77`) and set the path to `/usr/bin/tesseract`which is where my tesseract is installed to. But now, when I right-click on an entry and select "OCR selected PDF(s)" -> nothing happens.

Expected behaviour:
Something like: write the words or a set of the words in the textfield of the associated .pdf

Current behavour:
Nothing noticeable happens, CPU usage stays low.

I'm on Manjaro Linux (Arch derivative) with kernel `5.3.8` with tesseract `4.1.0`. My installed languages are `deu, eng, osd` and I tried to OCR an english pdf.

Am I doing something completely wrong?

Thanks in advance,
Lukas

**Comment by zuphilip** [comment]
Did you download the other prerequisite about `pdftoppm` and placed it in the Zotero directory to the other similar scripts? But we still need to make some lines more general in order to work also in Linux, e.g. https://github.com/UB-Mannheim/zotero-ocr/blob/master/chrome/content/zoteroocr.js#L14-L16 . @stweil Do you know whether these two lines with the `.exe` are the only ones which need to be changed?

I guess something like
```js
let pdfinfoScript = Zotero.isWin ? "pdfinfo.exe" : "pdfinfo";
let pdfinfo = dir.clone();
pdfinfo.append(pdfinfoScript);
```
and similar to pdftoppm should work on all platforms. Let me know when you test this or/and want to make a PR.

**Comment by luerhard** [comment]
I have pdftoppm version 0.80.0 installed, but I did not copy any files into a Zotero directory. 
I hardcoded my paths like:
```javascript
this.recognize = Zotero.Promise.coroutine(function* () {
		var ocrEngine = Zotero.Prefs.get("zoteroocr.ocrPath") || "/usr/bin/tesseract";
		alert(ocrEngine);
		...
		pdfinfo.append("pdfinfo");
		let pdftoppm = dir.clone();
		pdftoppm.append("pdftoppm");
```

I now create the following files in the folder of the pdf-Attachment:
- image-list.txt
- _filename_.info.txt
- _filename_.ocr.hocr (empty)
- _filename_.ocr.pdf (empty)
- _filename_.ocr.txt (empty)

The image-list.txt also contains a list of possible png-filepaths, but those files do not exist. Because I have never worked with javascript, can you point me to the line in the code where the image are supposed to be created?

**Comment by stweil** [comment]
Portability issues are addressed by the pull requests #5 and #6.

In addition, pull request #6 now adds reporting for missing executables (it shows an alert and aborts) and enters the found tesseract executable to the user settings. It also adds some debug messages which can help to find future problems.

**Comment by zuphilip** [comment]
This should be fixed now with version 0.2.0. Therefore I close this. If the problem still exists then feel free to open it again.
